src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -120,6 +139,9 @@ object Simplify {
 
       case GetTupleElement(MakeTuple(xs), idx) => xs(idx)
 
+      case ApplyIR("annotate", Seq(s1, MakeStruct(Seq())), _) =>
+        s1
```

Comment:
this is currently invalid. I had to remove this from a previous PR, because annotate and InsertFields make missing structs defined. We should change this, and then we can write this rule.